### PR TITLE
PAD-74: Add SBOM scan documentation

### PR DESCRIPTION
**Pull request by TylerGillson**
This PR documents the Software Bill of Materials (SBOM) capability added in 3.2.
💻 [Preview](https://deploy-preview-1108--docs-spectrocloud.netlify.app/clusters/cluster-management/compliance-scan#overview)
🎫 [Ticket](https://spectrocloud.atlassian.net/jira/software/c/projects/DOC/boards/62?modal=detail&selectedIssue=PAD-74)

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1108#" title="Author self-approved">TylerGillson</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by spectro-prow** [comment]
Welcome @TylerGillson! It looks like this is your first PR to spectrocloud/librarium 🎉

**Comment by spectro-prow** [comment]
Hi @TylerGillson. Thanks for your PR.

I'm waiting for a [spectrocloud](https://github.com/orgs/spectrocloud/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/spectrocloud/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: content/docs/04-clusters/06-cluster-management/03-compliance-scan.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/06-cluster-management/03-compliance-scan.md:66]:
```suggestion
#### SBOM Scan Format
```

File: content/docs/04-clusters/06-cluster-management/03-compliance-scan.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/06-cluster-management/03-compliance-scan.md:74]:
```suggestion
* [SPDX](https://github.com/spdx/spdx-spec/blob/v2.2/schemas/spdx-schema.json): A standardized format for representing SBOM that is widely used by organizations and governments. This report format has been around longer than any other SBOM format.


* [CycloneDX](https://cyclonedx.org/specification/overview/): An open-source XML format that provides a standard representation of software components and their metadata.


* GitHub's [dependency submission format](https://docs.github.com/en/rest/dependency-graph/dependency-submission?apiVersion=2022-11-28): Can be used to power a [dependency review workflow](https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-dependency-review).


* Syft JSON:  Syft's custom SBOM format. This format contains the most metadata compared to the other formats.
```
Updated the markdown list to include the descriptions within the same bullet.

File: content/docs/04-clusters/06-cluster-management/03-compliance-scan.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/06-cluster-management/03-compliance-scan.md:76]:
```suggestion
#### SBOM Scan Scopes
```

File: content/docs/04-clusters/06-cluster-management/03-compliance-scan.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/06-cluster-management/03-compliance-scan.md:84]:
```suggestion
* Cluster: Scans all the images in your Kubernetes cluster.


* Namespace: Scans all images in a particular Kubernetes namespace.


* Label Selector: Scans all images used by all of the Pods matching a label selector within a particular Kubernetes namespace.

* Pod: Scans all images used by a single Pod.
```

File: content/docs/04-clusters/06-cluster-management/03-compliance-scan.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/06-cluster-management/03-compliance-scan.md:86]:
```suggestion
## Review SBOM Results
```

File: content/docs/04-clusters/06-cluster-management/03-compliance-scan.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/06-cluster-management/03-compliance-scan.md:87]:
```suggestion
To review a completed scan, expand the **Software Bill of Materials (SBOM)** row. The expanded row displays the completed report containing detailed information about every scanned image. The context column indicates every unique usage of each image, broken out by container name, namespace, and pod name. Each image may be used by various containers within a given scope. The vulnerability summary column provides a condensed view of the vulnerability report, which can be viewed in greater detail by clicking on any row in the scan report.
```

File: content/docs/04-clusters/06-cluster-management/03-compliance-scan.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/06-cluster-management/03-compliance-scan.md:91]:
```suggestion
Each identified image has its own detailed results page. To review the image report, select the **>** button. Each image's details page within the scan report provides a list of dependencies and vulnerabilities.

The tables contained highlights of the metadata in the SBOM for a particular image. Each dependency's version and type are displayed. The amount of displayed details in the metadata section depends on the selected SBOM format. 
```

Flipped the order and clean up the language.

File: content/docs/04-clusters/06-cluster-management/03-compliance-scan.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/06-cluster-management/03-compliance-scan.md:95]:
```suggestion
For each identified vulnerability, you can view the name, severity level, vulnerability code, installed or impacted version, and the fix version (if a fix is available). Any CVEs documented in the [NIST National Vulnerability Database](https://nvd.nist.gov/vuln) (NVD) will render as a hyperlink to the NVD detail page for that particular vulnerability.
```

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1108#" title="Author self-approved">TylerGillson</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/release-3-2/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by karl-cardenas-coding — CHANGES_REQUESTED** [review]
@TylerGillson  thanks again for writing this up and the awesome images. Most of my changes are minor language clean up. Feel free to review the changes and ask questions. I'm gonna tag Rita to do the final copy edits.

File: content/docs/04-clusters/06-cluster-management/03-compliance-scan.md
Comment (ritawatson) [reply content/docs/04-clusters/06-cluster-management/03-compliance-scan.md:45]: In the last sentence, suggest: 
It details the various third-party components and dependencies used in the software and helps to manage security and compliance risks associated with those components.

File: content/docs/04-clusters/06-cluster-management/03-compliance-scan.md
Comment (ritawatson) [reply content/docs/04-clusters/06-cluster-management/03-compliance-scan.md:47]: comma in first sentence before such as can be removed because it introduces a comma list.

File: content/docs/04-clusters/06-cluster-management/03-compliance-scan.md
Comment (ritawatson) [reply content/docs/04-clusters/06-cluster-management/03-compliance-scan.md:91]: Suggest:
The details page for each image in the scan report provides a list of dependencies and vulnerabilities.

The tables list highlights of the metadata in the SBOM for a particular image. Each dependency's version and type are displayed. The amount of displayed details in the metadata section depends on the selected SBOM format.

File: content/docs/04-clusters/06-cluster-management/03-compliance-scan.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/03-compliance-scan.md:104]:
```suggestion
A cluster scan of any type can be started by navigating to the **Scans** tab of a cluster in Palette. Scan progress displays as 'Initiated' and transitions to 'Completed' when the scan is complete.
```

File: content/docs/04-clusters/06-cluster-management/03-compliance-scan.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/03-compliance-scan.md:111]:
```suggestion
You can set a schedule for each scan type when you deploy the cluster, and you can change the schedule at a later time.
```

File: content/docs/04-clusters/06-cluster-management/03-compliance-scan.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/03-compliance-scan.md:115]:
```suggestion
|Add New Cluster -> Cluster Policies -> Scan Policies -> Enable and schedule desired scans.|
```

File: content/docs/04-clusters/06-cluster-management/03-compliance-scan.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/03-compliance-scan.md:119]:
```suggestion
|Select the cluster to scan -> Settings -> Cluster Settings -> Scan Policies -> Enable and schedule scans of your choice.|
```

File: content/docs/04-clusters/06-cluster-management/03-compliance-scan.md
Comment (ritawatson) [inline content/docs/04-clusters/06-cluster-management/03-compliance-scan.md:123]:
```suggestion
Schedule your compliance scan for month, day, hour, or minute. For example: 
* Every week on Sunday at midnight.
* Every two weeks at midnight.
* Every month on the first day of the month at midnight.
* Every two months on the first day of the month at midnight
```

Comment (ritawatson) [reply content/docs/04-clusters/06-cluster-management/03-compliance-scan.md:123]: Made the examples a bullet list and removed the top-level bullet.

It would be useful to include an example of hour and minute to replace one of the week and one of the month examples.

**Review by ritawatson — APPROVED** [review]
Some minor suggestions.

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1108#pullrequestreview-1296738689" title="Approved">ritawatson</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1108#" title="Author self-approved">TylerGillson</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/release-3-2/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: content/docs/04-clusters/06-cluster-management/03-compliance-scan.md
Comment (TylerGillson) [reply content/docs/04-clusters/06-cluster-management/03-compliance-scan.md:91]: @karl-cardenas-coding @ritawatson one small issue with the re-wording(s) is that the additional metadata is in the SBOM - not on our UI. Users will only notice it if they download the SBOM and inspect it. Our UI shows the same info (dependencies & vulnerabilities) no matter what.

Comment (karl-cardenas-coding) [reply content/docs/04-clusters/06-cluster-management/03-compliance-scan.md:98]: @TylerGillson I see, in that case, it sounds like Rita's suggestion is probably closer to what you're thinking. We could end the paragraph by stating the additional metadata is only available in the downloaded format.

File: content/docs/04-clusters/06-cluster-management/03-compliance-scan.md
Comment (karl-cardenas-coding) [inline content/docs/04-clusters/06-cluster-management/03-compliance-scan.md:54]:
```suggestion
Palette will identify every unique container image within your chosen scope and generate an SBOM for that image. Palette also runs the SBOM through a vulnerability scanner to flag any Common Vulnerabilities and Exposures (CVEs). Palette leverages two open-source tools from Anchore: [Syft](https://github.com/anchore/syft) for SBOM generation and [Grype](https://github.com/anchore/grype) for vulnerability detection.
```

**Review by karl-cardenas-coding — APPROVED** [review]
Just a minor name typo, otherwise 👍🏻

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1108#pullrequestreview-1300010316" title="Approved">karl-cardenas-coding</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1108#pullrequestreview-1296738689" title="Approved">ritawatson</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1108#" title="Author self-approved">TylerGillson</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/release-3-2/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
